src/main.js

```diff
@@ -212,174 +108,36 @@ function allRelevantDataDirs() {
-        }
-
-        if (!keyFile.has_group(DISCOVERY_FEED_SECTION_NAME)) {
-            log('Key file ' + path + ' does not have a section called ' + DISCOVERY_FEED_SECTION_NAME + ', ignored');
             continue;
         }
 
-        let keys = keyFile.get_keys(DISCOVERY_FEED_SECTION_NAME)[0];
-        let requiredKeys = ['DesktopId', 'ObjectPath', 'BusName', 'SupportedInterfaces'];
-
-        let notFoundKeys = requiredKeys.filter(k => keys.indexOf(k) === -1);
-        if (notFoundKeys.length) {
-            log('Key file ' + path + ' does not have keys ' + notFoundKeys.join(', ') + ', ignoring');
-            continue;
-        }
-
-        let objectPath = null;
-        if (keyFile.has_group(LOAD_ITEM_SECTION_NAME)) {
-            log('Key file ' + path + ' does have a section called ' + LOAD_ITEM_SECTION_NAME + ', processing...');
-            try {
-                objectPath = keyFile.get_string(LOAD_ITEM_SECTION_NAME,
-                                                'ObjectPath');
-            } catch(e) {
-                log('Key file ' + path + ' does not have key \'ObjectPath\', ignoring');
-                continue;
-            }
-        }
-
-        let desktopId = maybeGetKeyfileString(keyFile,
-                                              DISCOVERY_FEED_SECTION_NAME,
-                                              'DesktopId',
-                                              null);
-
-        // Now, if we have a Desktop ID, we'll want to check it
-        // to see if there's an embedded language code in the
-        // desktop file. If so, filter out this application if it
-        // would not be compatible
-        if (desktopId) {
-            let providerLocale = appLanguage(desktopId);
-            if (providerLocale &&
-                !languageCodeIsCompatible(providerLocale, languages)) {
-                log('Language code ' + providerLocale + ' is not compatible ' +
-                    'with language codes ' + languages.join(' ') +
-                    ', skipping ' + path);
-                continue;
-            }
-        }
-
-        providerBusDescriptors.push({
-            path: keyFile.get_string(DISCOVERY_FEED_SECTION_NAME,
-                                     'ObjectPath'),
-            name: keyFile.get_string(DISCOVERY_FEED_SECTION_NAME,
-                                     'BusName'),
-            interfaces: keyFile.get_string(DISCOVERY_FEED_SECTION_NAME,
-                                           'SupportedInterfaces').split(';'),
-            knowledgeAppId: maybeGetKeyfileString(keyFile,
-                                                  DISCOVERY_FEED_SECTION_NAME,
-                                                  'AppID',
-                                                  null),
-            desktopFileId: desktopId,
-            knowledgeSearchObjectPath: objectPath
-        });
-    }
+        // Now that we have the keyfile, set the Exec line to some
```

Comment:
Why do we still need this code? We don't want to have two copies of it, one in JS and one in C...

Reply:
(Still need to address this)